src/Grocery/Grocery.js

```diff
@@ -1,34 +1,131 @@
 import React, { Component } from 'react';
-import { CheckBox } from 'react-native-elements';
-
 import Meteor from 'react-native-meteor';
 
+import { colors, stylesheet } from '../../config/styles';
+
+import { TextInput, SafeAreaView, Alert, Text } from 'react-native';
+import { Button } from 'react-native-elements';
+
 export default class Grocery extends Component {
+	
 	constructor(props) {
 		super(props);
+		
+		const groceryId = props.navigation.state.params.id;
+		let name = '';
+		let amount = null;
+		let newGrocery = true;
+		if (groceryId) {
+			newGrocery = false;
+
+			const grocery = this.props.screenProps.groceries.find(grocery => grocery._id === groceryId);
+			if (grocery) {
+				name = grocery.name;
+				amount = grocery.amount;
+			}
+		}
+		this.state = {
+			listId: props.navigation.state.params.listId,
+			name,
+			amount,
+			newGrocery
+		}
 	}
+    
+	addGrocery() {
+		if (this.state.name.length === 0) {
+			Alert.alert(
```

Comment:
Not sure if it's just on my Mac, but the Alert makes the iOS simulator super laggy for me 😕

Reply:
It's working fine for me on my Mac